src/transformers/modeling_t5.py

```diff
@@ -916,3 +916,22 @@ def forward(self, **kwargs):
             ) + decoder_outputs  # TODO(thom): Add z_loss https://github.com/tensorflow/mesh/blob/fa19d69eafc9a482aff0b59ddd96b025c0cb207d/mesh_tensorflow/layers.py#L666
 
         return decoder_outputs + encoder_outputs
+
+    def prepare_inputs_for_generation(self, input_ids, past, attention_mask, **kwargs):
+        assert past is not None, "past has to be defined for encoder_outputs"
+
+        # first step
+        if type(past) is tuple:
+            encoder_hidden_states = past[0]
+        else:
+            encoder_hidden_states = past
+
+        return {
+            "decoder_input_ids": input_ids,
+            "encoder_hidden_states": encoder_hidden_states,
+            "encoder_attention_mask": attention_mask,
+        }
+
+    def _reorder_cache(self, past, beam_idx):
+        # past does not have to be reorderd for T5.
```

Comment:
@craffel why? How do you associate beam_hypotheses[i] and hidden_states[i]?

Reply:
(nit) typo: reordered

Reply:
@sshleifer this was actually me who orded this. As I implemented it right now, `past` are just the `encoder_hidden_states` ( `encoder_outputs`) and they don't change for different `beam_idx`